src/plugins/data_source_management/public/components/data_source_selector/data_source_selector.tsx

```diff
@@ -57,66 +58,130 @@ export class DataSourceSelector extends React.Component<
     this.state = {
       allDataSources: [],
       defaultDataSource: '',
-      selectedOption: this.props.defaultOption
-        ? this.props.defaultOption
-        : this.props.hideLocalCluster
-        ? []
-        : [LocalCluster],
+      selectedOption: this.props.hideLocalCluster ? [] : [LocalCluster],
     };
   }
 
   componentWillUnmount() {
     this._isMounted = false;
   }
 
-  async componentDidMount() {
-    this._isMounted = true;
+  handleSelectedOption(
+    dataSourceOptions: DataSourceOption[],
+    allDataSources: Array<SavedObject<DataSourceAttributes>>,
+    defaultDataSource: string | null
+  ) {
+    const [{ id }] = this.props.defaultOption!;
+    const dataSource = dataSourceOptions.find((ds) => ds.id === id);
+    const selectedOption = dataSource ? [{ id, label: dataSource.label }] : [];
+
+    // Invalid/filtered out datasource
+    if (!dataSource) {
+      this.props.notifications.addWarning(
+        i18n.translate('dataSource.fetchDataSourceError', {
+          defaultMessage: 'Data source with id is not available',
+        })
+      );
```

Comment:
Should we also use the callback function to tell plugins team that Invalid state?

Reply:
So this function, regardless of valid/invalid id, will set the new state and call the callback function. If the id is valid, we change the `selectedOption` to be whatever was set in props and call the callback function with this. IF the id is invalid, we change the `selectedOption` to be `[]` and return the callback function with `[]`. The only thing that changes is whether to add a toast message.

Reply:
GOt it.